### DISCUSSION TITLE
Kenapa fungsi reorder di ggplot tidak jalan

### DISCUSSION THREAD

**Discussion by 19807311** [teamalgoritma/community#421] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/114885734/198578674-20445093-3b63-4ca6-bff2-01b2f2ba81c1.png)

Halo, mau bertanya saya mau membuat plot ini untuk memvisualisasikan proportion. Kenapa fungsi reorder di ggplpt (aes) tidak jalan ya sehingga tampilan proporsinya tidak berurutan begitu. 
Adakah cara lainnya?

Terima kasih

**Comment by VictorNugraha** [comment, marked as answer]
Hai, @19807311 !

Untuk mengurutkan _value_ pada _stacked bar plot_ tidak bisa menggunakan fungsi **_reorder()_** karena fungsi tersebut hanya akan membandingkan keseluruhan _value_ yang terdapat pada kolom _category_.

Ibu bisa mencoba untuk menambahkan fungsi **_fct_reorder()_** pada parameter **_fill =_**. 

- Intuisi penggunaan fungsi **_fct_reorder()_** sama seperti fungsi **_reorder()_**, dimana fungsi tersebut dapat kita isi dengan nama kolom yang ingin kita urutkan dan kita urutkannya berdasarkan nilai apa.
- Intuisi kenapa kita mengurutkan pada parameter **_fill =_**, karena yang membedakan informasi _Sub Category_ adalah warna pada _bar plot_ tersebut.

Berikut contoh syntax:

```
ggplot(data = df, 
       aes(x = Category, 
           y = sum_qty, 
           fill = fct_reorder(Sub.Category, sum_qty))) + 
geom_bar()
```

Demikian informasi yang dapat saya sampaikan, selamat mencoba dan semoga bermanfaat.
